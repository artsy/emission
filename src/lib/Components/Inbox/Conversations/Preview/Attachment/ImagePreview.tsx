import * as React from "react"
import { TouchableHighlight } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import styled from "styled-components/native"

import colors from "../../../../../../data/colors"
import fonts from "../../../../../../data/fonts"

import OpaqueImageView from "../../../../OpaqueImageView"
import AttachmentPreview, { AttachmentProps } from "./AttachmentPreview"

const Image = styled(OpaqueImageView)`
  height: 150;
  flex: 1;
`

interface Props extends AttachmentProps, RelayProps {}

export const ImagePreview: React.SFC<Props> = ({ attachment, onSelected }) =>
  <AttachmentPreview attachment={attachment as any} onSelected={onSelected}>
    <Image imageURL={attachment.download_url} />
  </AttachmentPreview>

export default createFragmentContainer(
  ImagePreview,
  graphql`
    fragment ImagePreview_attachment on Attachment {
      download_url
      ...AttachmentPreview_attachment
    }
  `
)

interface RelayProps {
  attachment: {
    download_url?: string
  }
}
