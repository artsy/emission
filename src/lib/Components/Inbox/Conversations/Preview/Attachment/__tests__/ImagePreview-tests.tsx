import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import { ImagePreview } from "../ImagePreview"

it("renders correctly", () => {
  const tree = renderer.create(<ImagePreview attachment={attachment} />)
  expect(tree).toMatchSnapshot()
})

const attachment = {
  id: "cats",
  download_url: "/path/to/cats.jpg",
}
