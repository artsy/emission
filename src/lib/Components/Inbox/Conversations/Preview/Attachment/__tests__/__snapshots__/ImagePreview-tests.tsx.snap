// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        null,
      ]
    }
  >
    <AROpaqueImageView
      imageURL="/path/to/cats.jpg"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 150,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
